litestar/plugins.py

```diff
@@ -148,6 +151,36 @@ def from_data_container_instance(
         raise NotImplementedError()
 
 
+@runtime_checkable
+class DTOSerializationPluginProtocol(Protocol):
+    """Protocol used to define a serialization plugin for DTOs."""
+
+    __slots__ = ()
+
+    @staticmethod
+    def supports_type(parsed_type: ParsedType) -> bool:
+        """Given a value of indeterminate type, determine if this value is supported by the plugin.
+
+        Args:
+            parsed_type: A parsed type.
+
+        Returns:
+            Whether the type is supported by the plugin.
+        """
+        raise NotImplementedError()
+
+    def create_dto_for_type(self, parsed_type: ParsedType) -> type[AbstractDTOFactory]:
```

Comment:
```suggestion
    def create_dto_for_type(self, parsed_type: ParsedType) -> type[DTOInterface]:
```
Protocol shouldn't care what interface implementation it is.